### PR TITLE
Final version of profile overview

### PR DESCRIPTION
**Pull request by olga-kolosek**
- View user's profile

- Edit email, password, full name, bio, role (for admin)

### CODE REVIEW

File: MyApp/app/controllers/users/registrations_controller.rb [hunk 1]
```diff
@@ -2,27 +2,36 @@
...
-  #   super
-  # end
+ def create
+    super
+    users = User.all
+    if users.length == 1
+      @user.add_role(:admin)
+    else
+      @user.add_role(:author)
+    end
```
Comment (tamara-bol) [inline MyApp/app/controllers/users/registrations_controller.rb]: It's fine, but you can make it shorter:
User.all.length == 1 ? @user.add_role(:admin) : @user.add_role(:author)

File: MyApp/app/controllers/users/registrations_controller.rb [hunk 2]
```diff
@@ -2,27 +2,36 @@
...
+    @user.save
+  end
 
   # GET /resource/edit
-  # def edit
-  #   super
-  # end
+  def edit
+    super
+  end
```
Comment (tamara-bol) [inline MyApp/app/controllers/users/registrations_controller.rb:17]: remove what you don't actually overriding

File: MyApp/db/migrate/20201210120233_add_role_to_users.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddRoleToUsers < ActiveRecord::Migration[6.0]
+  def change
+  	   add_column :users, :role, :string
```
Comment (tamara-bol) [inline MyApp/db/migrate/20201210120233_add_role_to_users.rb:3]: lets make it integer and use enum

like here for example: https://www.justinweiss.com/articles/creating-easy-readable-attributes-with-activerecord-enums/

**Review by tamara-bol — CHANGES_REQUESTED** [review]

File: MyApp/app/controllers/application_controller.rb [hunk 4]
```diff
@@ -13,6 +13,6 @@ class ApplicationController < ActionController::Base
 
   def configure_permitted_parameters
     devise_parameter_sanitizer.permit(:sign_up, keys: [:name])
-    devise_parameter_sanitizer.permit(:account_update, keys: [:name])
+    devise_parameter_sanitizer.permit(:account_update, keys: [:name, :image, :full_name, :role, :bio])
```
Comment (tamara-bol) [inline MyApp/app/controllers/application_controller.rb]: You should permit this params is user_controller, and also, do update for user in users_controler, not registraton, 
same for show action, which I dont see but description says: View user's profile

**Review by tamara-bol — CHANGES_REQUESTED** [review]

File: MyApp/app/controllers/application_controller.rb (hunk 4 above)
Comment (olga-kolosek) [reply MyApp/app/controllers/application_controller.rb]: I didn't understand this quite right. I am using devise gem for authentication and in the tutorial, they advised to place all methods regarding user (new / edit..) in registrations_controller, so I followed the comments.

Comment (tamara-bol) [reply MyApp/app/controllers/application_controller.rb]: Where is this advise that you have followed?

**Review by tamara-bol — APPROVED** [review]

File: MyApp/app/controllers/application_controller.rb (hunk 4 above)
Comment (olga-kolosek) [reply MyApp/app/controllers/application_controller.rb]: Here it is: https://github.com/heartcombo/devise

**Pull request merged**